src/App.tsx

```diff
@@ -1,61 +1,94 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
+
+export const App: React.FC = () => {
+  const [currentTitle, setCurrentTitle] = useState('');
+  const [currentUserId, setCurrentUserId] = useState(0);
+  const [todos, setTodos] = useState(todosFromServer);
+  const [wasSubmitted, setWasSubmitted] = useState(false);
+
+  const isFormValid = currentTitle.trim() && currentUserId;
+
+  const formReset = () => {
+    setCurrentTitle('');
+    setCurrentUserId(0);
+  };
+
+  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (isFormValid) {
+      const newTodo = {
+        id: Math.max(...todos.map(todo => todo.id)) + 1,
+        title: currentTitle,
+        userId: currentUserId,
+        completed: false,
+        user: usersFromServer.find(user => user.id === currentUserId) || '',
+      };
+
+      setTodos([...todos, newTodo]);
+
+      formReset();
+    } else {
+      setWasSubmitted(true);
+    }
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={onSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="titleInput">Title: </label>
+          <input
+            type="text"
+            data-cy="titleInput"
+            name="title"
+            id="titleInput"
+            value={currentTitle}
+            onChange={e => setCurrentTitle(e.target.value)}
+            placeholder="Enter a title"
+          />
+          {wasSubmitted && !currentTitle && (
+            <span className="error">Please enter a title</span>
+          )}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
+          <label htmlFor="userSelect">User: </label>
+          <select
+            data-cy="userSelect"
+            name="user"
+            id="userSelect"
+            value={currentUserId}
+            onChange={e => setCurrentUserId(+e.target.value)}
```

Comment:
[CODE STYLE] - Don't use one-letter variable naming. It could be confusing for another developer who reads your code. Consider renaming 'e' to 'event' or another descriptive name.